pkgs/development/mobile/androidenv/tools.nix

```diff
@@ -1,26 +1,42 @@
-{deployAndroidPackage, requireFile, lib, packages, toolsVersion, os, callPackage, postInstall ? ""}:
+{deployAndroidPackage, lib, package, autoPatchelfHook, makeWrapper, os, pkgs, pkgsi686Linux, postInstall}:
 
-if toolsVersion == "26.0.1" then callPackage ./tools/26.nix {
-  inherit deployAndroidPackage lib os postInstall;
-  package = {
-    name = "tools";
-    path = "tools";
-    revision = "26.0.1";
-    archives = {
-      linux = requireFile {
-        url = "https://dl.google.com/android/repository/sdk-tools-linux-3859397.zip";
-        sha256 = "185yq7qwxflw24ccm5d6zziwlc9pxmsm3f54pm9p7xm0ik724kj4";
-      };
-      macosx = requireFile {
-        url = "https://dl.google.com/android/repository/sdk-tools-darwin-3859397.zip";
-        sha256 = "1ycx9gzdaqaw6n19yvxjawywacavn1jc6sadlz5qikhgfr57b0aa";
-      };
-    };
-  };
-} else if toolsVersion == "26.1.1" then callPackage ./tools/26.nix {
-  inherit deployAndroidPackage lib os postInstall;
-  package = packages.tools.${toolsVersion};
-} else callPackage ./tools/25.nix {
-  inherit deployAndroidPackage lib os postInstall;
-  package = packages.tools.${toolsVersion};
+deployAndroidPackage {
+  name = "androidsdk";
+  inherit os package;
+  nativeBuildInputs = [ makeWrapper ]
+    ++ lib.optionals (os == "linux") [ autoPatchelfHook ];
+  buildInputs = lib.optional (os == "linux") (
+      (with pkgs; [ glibc freetype fontconfig fontconfig.lib])
+      ++ (with pkgs.xorg; [ libX11 libXrender libXext ])
+      ++ (with pkgsi686Linux; [ glibc xorg.libX11 xorg.libXrender xorg.libXext fontconfig.lib freetype zlib ])
+    );
+
+  patchInstructions = ''
+    ${lib.optionalString (os == "linux") ''
+      # Auto patch all binaries
+      autoPatchelf .
+    ''}
+
+    # Wrap all scripts that require JAVA_HOME
+    for i in bin; do
+      find $i -maxdepth 1 -type f -executable | while read program; do
+        if grep -q "JAVA_HOME" $program; then
+          wrapProgram $PWD/$program --prefix PATH : ${pkgs.jdk8}/bin
+        fi
+      done
+    done
+
+    # Wrap monitor script
+    wrapProgram $PWD/monitor \
+      --prefix PATH : ${pkgs.jdk8}/bin \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ pkgs.xorg.libX11 pkgs.xorg.libXtst ]}
+
+    # Patch all script shebangs
+    patchShebangs .
```

Comment:
Can't we limit this to a subdirectory?

Reply:
First of all I have to mention that `tools` is obsolete so I didn't change its code. I just copied the entire file from `tools/26.nix` to here. Second this is its structure:

```
dr-xr-xr-x 1 root root    216 Dec 31  1969 .
dr-xr-xr-x 1 root root     68 Dec 31  1969 ..
-r-xr-xr-x 1 root root   1313 Dec 31  1969 .monitor-wrapped
-r--r--r-- 1 root root 829319 Dec 31  1969 NOTICE.txt
-r-xr-xr-x 1 root root   4909 Dec 31  1969 android
dr-xr-xr-x 1 root root    420 Dec 31  1969 bin
-r-xr-xr-x 1 root root 629936 Dec 31  1969 emulator
-r-xr-xr-x 1 root root 414688 Dec 31  1969 emulator-check
dr-xr-xr-x 1 root root   3478 Dec 31  1969 lib
-r-xr-xr-x 1 root root  12191 Dec 31  1969 mksdcard
-r-xr-xr-x 1 root root   1159 Dec 31  1969 monitor
dr-xr-xr-x 1 root root    216 Dec 31  1969 proguard
-r--r--r-- 1 root root    138 Dec 31  1969 source.properties
dr-xr-xr-x 1 root root    222 Dec 31  1969 support
```
As you can see executables are all over the place so it looks like `patchShebangs .` is correct.